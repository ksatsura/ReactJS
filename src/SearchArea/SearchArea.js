//imports
import React from 'react';
import SearchFilter from '../SearchFilter/SearchFilter';
import { connect } from 'react-redux';
import { saveInputValue } from '../actions/searchActions';
import '../style.css';

export const SearchArea = (props) => {

    const { value, handleChange, match, location } = props;

    const paramValue = match && match.params.value;

    return (
        <div className='search-area'>
            <h1 className='search-title'>find your movie</h1>
            <input type="search" name="search-value" className='search-value' value={value} onChange={(e) => handleChange(e.target.value)} />
            <SearchFilter location={location} paramValue={paramValue} />
        </div>
    );  
};

const mapDispatchToProps = dispatch => {
    return {
        handleChange: (value) => dispatch(saveInputValue(value)),
    };
};

const mapStateToProps = state => {
    return {
        value: state.filmListReducer.currentValue,
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchArea);