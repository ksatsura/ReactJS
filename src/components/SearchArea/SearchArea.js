//imports
import * as React from 'react';
import SearchFilter from '../SearchFilter/SearchFilter';
import { connect } from 'react-redux';
import { saveInputValue } from '../../actions/searchActions';
import injectSheet from 'react-jss';
import '../../style.css';

// @flow
type Props = {
    value: string, 
    handleChange: (string) => void, 
    match: {
        params: {
            value: string
        }
    },
    location: {},
};

const styles = {
    searchArea: {
        display: 'flex',
        flexDirection: 'column',
    },
    searchTitle: {
        color: 'white',
        margin: {
            top: 10,
            right: 60,
            bottom: 10,
            left: 60
        },
        textTransform: 'uppercase',
    },
    searchValue: {
        height: 40,
        width: 1080,
        fontSize: 30,
        backgroundColor: 'rgba(0, 0, 0, 0.9)',
        margin: {
            top: 10,
            right: 60,
            bottom: 20,
            left: 60
        },
        color: 'white',
        'border-top': [2, 'solid', 'black'],
        'border-right': [2, 'solid', 'black'],
        'border-bottom': [2, 'solid', '#ff0055'],
        'border-left': [2, 'solid', 'black'],
    },
};
  
const SearchAreaDiv = ({ classes, value, handleChange, children }) => (
    <div className={classes.searchArea}>
        <h1 className={classes.searchTitle}>find your movie</h1>
        <input type="search" name="search-value" className={classes.searchValue} value={value} onChange={(e) => handleChange(e.target.value)} />
        { children }
    </div>
);

const StyledSearchAreaDiv = injectSheet(styles)(SearchAreaDiv);

export const SearchArea = (props: Props) => {

    const { value, handleChange, match, location } = props;

    const paramValue = match && match.params.value;

    return (
        <StyledSearchAreaDiv value={value} handleChange={handleChange} >
            <SearchFilter location={location} paramValue={paramValue} />
        </StyledSearchAreaDiv>
    );  
};

const mapDispatchToProps = dispatch => {
    return {
        handleChange: (value) => dispatch(saveInputValue(value)),
    };
};

const mapStateToProps = state => {
    return {
        value: state.filmListReducer.currentValue,
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchArea);