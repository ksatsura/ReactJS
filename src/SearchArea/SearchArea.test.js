import React from 'react';
import SearchArea from './SearchArea.js';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';

describe('SearchArea component', () => {

    let handleClick;

    beforeAll(() => {
        handleClick = jest.fn();
    });

    it('should be render correctly', () => {
        const component = shallow(<SearchArea handleClick={handleClick} />);
        
        expect(component).toMatchSnapshot();
    });

    it('test search button click event', () => {
        const component = shallow(<SearchArea handleClick={handleClick} />);
        
        component.find('.search-button').simulate('click');
        expect(handleClick.mock.calls.length).toEqual(1);
    });

    it('test submit button click event', () => {
        const fakeEvent = { preventDefault: () => {} };
        const component = shallow(<SearchArea handleClick={handleClick} />);
        component.instance().handleSubmit = jest.fn();
        
        component.find('.search-form').simulate('submit', fakeEvent);
        expect(component.instance().handleSubmit).toHaveBeenCalled;
    });

    it('test search field change event', () => {
        const component = shallow(<SearchArea handleClick={handleClick} />);

        component.instance().handleChange = jest.fn();
        component.update();
        component.instance().handleChange('the-value');
        expect(component.instance().handleChange).toBeCalledWith('the-value');
    });
});
