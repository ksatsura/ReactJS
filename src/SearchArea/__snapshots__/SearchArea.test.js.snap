// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchArea component should be render correctly 1`] = `ShallowWrapper {}`;
